Initial stab at welcome message (#1)
for-the-horde/greet-bot: WELCOME.md

> 👋 Hello @new_user_name and welcome to the For the Horde community! We're so happy you're here! The purpose of this community is to give you a space to connect with other Minnesota-based tech and tech-adjacent folks. Want to be successful here? Give more than you take.

Reviewer: ```suggestion
👋 Hello @new_user_name and welcome to the For the Horde community! We're so happy you're here! The purpose of this community is to give you a space to connect with other Minnesota-based tech and tech-adjacent folks. Want to be successful here? Try to give more than you take.
```

> ✅ First things first. This is a community and we want this to be a safe and inclusive space for everyone. We expect you to abide by the For the Horde Code of Conduct in this Slack team AND while participating in any in-person or online For the Horde events. Please head on over to [http://git.io/v0d6b](http://git.io/v0d6b).

Reviewer: ```suggestion
✅ First things first. This is a community and we want this to be a safe and inclusive space for everyone. We expect you to abide by the For the Horde Code of Conduct in this Slack team AND while participating in any in-person or online For the Horde events. Please head on over to [http://git.io/v0d6b](http://git.io/v0d6b) to check it out.
```

Author: I also wonder if we should require people to check a box, like in a google form. Not that we have to check it all the time, but to have a record.

Thoughts?

Reviewer: That's a cool idea! I was overengineering and asking myself if I could eventually write a slack app that would deliver this message and have an interactive element for the user to click to record acknowledgement, but a google form is awesome because it gets the job done now.

Need to report Code of Conduct violation? The admins are listed at the bottom of the Code of Conduct. Snap a screenshot, report it.

> 🚴🏿‍♀️🕺🏻👩‍💻😻⚾️✈️📺 Second, you've automagically been added to the #introductions channel. Please introduce yourself to the community there and share something fun about yourself. Where you're from, what you're working on, what your most peculiar hobby is, what your favorite color is, etc.. We want to get to know you! Also, there's probably a channel for that.

Reviewer: ```suggestion
🚴🏿‍♀️🕺🏻👩‍💻😻⚾️✈️📺 Second, you've automagically been added to the #introductions channel. Please introduce yourself to the community there and share something fun about yourself. Where you're from, what you're working on, what your most peculiar hobby is, what your favorite color is, etc... We want to get to know you! Also, there's probably a channel for that peculiar hobby.
```
Took me a while to get the last sentence, so I added the clarification

> 📣 Speaking of channels. We have a lot. When thinking about where to post a message, go through the channel list and check there first. It’s very tempting with ~600 eyeballs to post something to #general. We tend to treat #general like a watercooler, not a place to garner attention.

Reviewer: ```suggestion
📣 Speaking of channels. We have a lot. When thinking about where to post a message, go through the channel list and check there first. It’s very tempting with ~1200 eyeballs to post something to #general. We tend to treat #general like a watercooler, not a place to garner attention.
```
I'm gonna estimate the average number of eyeballs per member to be ~2 😂

Author: hahaha yep!

_Popular channels:_

- #frontend - talk about your javascript

> - #devs-n-ops - do not talk about your javascript

Reviewer 2: I think we can re-word the #frontend and #dev-n-ops descriptions so they read as more _inclusive_ and less snarky (no offense Jake).

Here's my stab at it:
```
- #devs-n-ops - a safe place to talk about development & ops related stuff (read: not front-end javascript)
- #frontend - a safe place to talk about your javascript :face_with_rolling_eyes:
```

Reviewer: I hesitated at the #devs-n-ops description too, but didn't suggest a change because fun and personality in this message is valuable. I'm still not opposed to updating the #devs-n-ops description, but think the light/joke-y tone is a feature and not a bug.

Author: I'm going to noodle on this more. I don't think `:face_with_rolling_eyes:` fixes this, but I also understand where you're coming from. One thing I also considered (because I wanted to ask in one of these channels about it the other day and didn't know where to put it) is that javascript IS a backend technology.

- #cohordeworking - when, where, why, who
- #business - general business topics
- #mental-health - talk about it

> - #gigs - the ONLY place you can post/ask for jobs/gigs. They should be 100% available to Minnesota with no filter criteria, ala “might be interested in remote” Many people have notifications on this channel, so 🧵any conversations beyond the initial post.

Reviewer: ```suggestion
- #gigs - the ONLY place you can post/ask for jobs/gigs. They should be 100% available to Minnesota with no filter criteria, à la “might be interested in remote” Many people have notifications on for this channel, so 🧵any conversations beyond the initial post.
```

> 

Reviewer 2: Based on channel popularity, I think it'd be good to include #event-share and #howbazaar in this list, (maybe above #gigs because it's description is long)

```
#event-share: share local COOL events
#howbazaar: buy/sell/trade classifieds
```

Author: I think adding #event-share is a good idea, but I don't think it's necessary to add #howbazaar. It might be in the top channel list because of number of users, but I don't think it's critical to know about it.

_Fun, small channels:_

- #art-hackers
- #biking
- #awww
- #pets
- #sports

> - #outdoors

Reviewer 2: #gaming
#nerdparenting

Reviewer: imo, these are by design a sampling and not exhaustive. The welcome message is already dangerously long and risks being ignored, so I would actually consider shortening these lists before adding to them...

Reviewer 2: Not trying to be exhaustive, and I agree with you that this is already very long. But are two additional items in a bulleted list going to affect whether someone reads the whole thing or not? My guess is that they'll quickly scan the list regardless, and the presence of two additional topics not mentioned anywhere else in here _might_ encourage them to mention that they have kids or are gamers in their intro.

Author: Ok, I'm going to remove #outdoors and #pets and add #nerdparenting. #gaming, #pets, and #outdoors can be discoveries.

> 

Reviewer 2: Add #gaming to the list

🦠 And then there’s COVID-19, the big 🐘 in the room. #general is a COVID-19 free zone. This is a **strict** policy and we will actively police it. While we understand it’s a hard time for everyone, you can’t leave #general and we want people to have a refuge as it is impacting many of us indirectly or directly.

SO with that:
#covid19-discuss - general discussion of COVID-19 related topics
#covid19-reference - fact based information sharing (links only, move discussions to #covid19-discuss)

> #covid19-memes - the ONLY place to allow covid related memes (that also adhere to the Code of Conduct)

Reviewer 2: I'll set a slack reminder in the admins channel for 6 months out that reminds us to remove the covid section in the welcome message (if these channels are no longer necessary) 🤞 😬

Reviewer 2: Reminder set!

Author: Thanks!

Author: Your reminder should probs say "Remove COVID stuff and add back in monthly IRL gatherings"

Reviewer 2: Eeeee good point

> :spam: Spam posts, sales-y posts, self-promoting without community participation is not allowed. Admins will ask posts like this to be removed or moved to a different channel. If not done by the original poster within 15 min, we will do it for you. Don't make us do it for you.

Reviewer: ```suggestion
:spam: Spam posts, sales-y posts, self-promoting without community participation is not allowed. For the Horde is a community to participate in, not an audience to broadcast to. Admins will ask posts like this to be removed, especially in #general. If not removed promptly, we will remove it. If you're really sure the post benefits the audience more than yourself, be sure to put it in the most relevant channel.
```
Riffing here, which is kind of awkward for prose in github. Aims I had with these changes:
* remove specific time limit (puts admins on a long clock monitoring the situation, and promises bad actors their spam will live for at least 15 minutes)
* introduce the community vs audience concept (I've become really attached to that language because I think it succinctly says a lot, in a conversation where "spam" and "promotion" can be hotly debated
* try to set a criteria for whether something is too spammy/promotional (does it benefit the audience more than the poster)

Author: yes, i love this!

💌 Want to invite someone? Head over to #invites and kindly answer these questions:

> - Who are you inviting? (name and email)

Reviewer: ```suggestion
- Who would you like to invite? (name and email)
```
Removes presumption of approval

> :spam: Spam posts, sales-y posts, self-promoting without community participation is not allowed. For the Horde is a community to participate in, not an audience to broadcast to. Admins will ask posts like this to be removed, especially in #general. If not removed promptly, we will remove it. If you're really sure the post benefits the audience more than yourself, be sure to put it in the most relevant channel.

Reviewer 2: This is nit picky, and maybe it's just me, but I'd like to italicize _to participate in_ (and maybe _broadcast to_ also)

Reviewer: IANAD, but maybe bold "participate in" and italicize "broadcast to"? Emphasize participate and diminish broadcast?

Author: Cool.

So @jcardinal... IANAD means "I am not a doctor"?!?

Reviewer: @jennapederson I was going for "I am not a designer" 😆

- Do they currently reside in Minnesota?

> - Describe how they are tech adjacent or tech in general.

Reviewer: ```suggestion
- Describe how they are tech or tech-adjacent.
```
Lead with tech, tech-adjacent secondary

> - Are they expecting this invite? We only invite people who have opted into this invite, otherwise we're obnoxious.

Reviewer: ```suggestion
- Are they expecting this invite? We only invite people who have opted in, otherwise we're obnoxious.
```

> _Note:_ Please invite good humans that will read and abide by the Code of Conduct. You're using _your_ social capital to say "this is a good human."

Reviewer: This is awesome

Author: The first sentence is @whoisjake's. The second is mine 😊

Reviewer: They're both great, and both important!

Reviewer 3: I like it.

> ❓Ask questions! It’s ok to not know how things work here or if things are appropriate.

Reviewer: ```suggestion
❓Ask questions! It's good to read the room, but it's a big room with norms that have formed over a long time. It’s ok if you don't know how things work here or what things are appropriate.
```
